src/CombFilter/CombFilter.h

```diff
@@ -0,0 +1,56 @@
+#if !defined(__CombFilter_hdr__)
+#define __CombFilter_hdr__
+
+#include "ErrorDef.h"
+
+#include "CombFilterIf.h"
+#include "RingBuffer.h"
+
+
+class CCombFilterBase : public CCombFilterIf
+{
+public:
+	CCombFilterBase(float fMaxDelayLengthInS, float fSampleRateInHz, int iNumChannels);
+	virtual ~CCombFilterBase();
+
+	virtual Error_t process(float** ppfAudioInputBuffer, float** ppfAudioOutputBuffer, int iNumberOfFrames) = 0;
```

Comment:
I especially like the pure virtual function definition here, which refuses others from initializing the base class that actually does not contain any signal processing functionality.